test/Cli/Func.UnitTests/ActionsTests/StartHostActionTests.cs

```diff
@@ -261,6 +268,42 @@ public async Task ValidateHostRuntimeAsync_MatchesExpectedResults(WorkerRuntime
             Assert.False(expectException, "Expected validation failure.");
         }
 
+        [InlineData(WorkerRuntime.DotnetIsolated)]
+        [InlineData(WorkerRuntime.Python)]
+        [InlineData(WorkerRuntime.Java)]
+        [InlineData(WorkerRuntime.Node)]
+        [Theory]
+        public async Task AzureFunctionsEnvironment_EnvironmentVariable_SetByUser_DoesNotThrow(WorkerRuntime currentRuntime)
+        {
+            Environment.SetEnvironmentVariable(Common.Constants.FunctionsWorkerRuntime, WorkerRuntimeLanguageHelper.GetRuntimeMoniker(currentRuntime));
+            GlobalCoreToolsSettings.SetWorkerRuntime(currentRuntime);
+
+            Mock<IProcessManager> processManager = new();
+            Mock<ISecretsManager> secretsManager = new();
+
+            var settings = new Dictionary<string, string> { ["AZURE_FUNCTIONS_ENVIRONMENT"] = "MyEnvironment" };
+            secretsManager.Setup(s => s.GetSecrets()).Returns(() => settings);
+
+            GlobalCoreToolsSettings.Init(secretsManager.Object, []);
+
+            var startHostAction = new StartHostAction(secretsManager.Object, processManager.Object)
```

Comment:
My bad, I just noticed that you're actually running the CLI with this test. So it's not a unit test but would need to use the E2E test framework we have. Here is an example of how to write one of those instead:

https://github.com/Azure/azure-functions-core-tools/blob/e963a7607b71a06d8cd2780833c5487fa50c511c/test/Cli/Func.E2ETests/Commands/FuncStart/MissingConfigTests.cs#L50-L83

I believe there is a [doc](https://github.com/Azure/azure-functions-core-tools/blob/main/test/Cli/Func.E2ETests/README.md) too